service-tests/github.js

```diff
@@ -9,13 +9,50 @@ const {
   isFormattedDate,
   isVPlusDottedVersionAtLeastOne
 } = require('./helpers/validators');
+const colorscheme = require('../lib/colorscheme.json');
 
 const t = new ServiceTester({ id: 'github', title: 'Github' });
 module.exports = t;
-
-t.create('License')
-  .get('/license/badges/shields.json')
-  .expectJSONTypes(Joi.object().keys({ name: 'license', value: Joi.string() }));
+const colorsB = Object.assign({}, ...Object.keys(colorscheme).map(color => ({ [color]: colorscheme[color].colorB })));
+
+t.create('Public domain license')
+  .get('/license/badges/shields.json?style=_shields_test')
+  .expectJSON({ name: 'license', value: 'CC0-1.0', colorB: '#7cd958' });
+
+t.create('Copyleft license')
+  .get('/license/ansible/ansible.json?style=_shields_test')
+  .expectJSON({ name: 'license', value: 'GPL-3.0', colorB: colorsB.orange });
+
+t.create('Permissive license')
+  .get('/license/atom/atom.json?style=_shields_test')
+  .expectJSON({ name: 'license', value: 'MIT', colorB: colorsB.green });
+
+t.create('License for repo without a license')
+  .get('/license/badges/badger.json?style=_shields_test')
+  .expectJSON({ name: 'license', value: 'missing', colorB: colorsB.red });
+
+t.create('License for repo with an unrecognized license')
+  .get('/license/philokev/sopel-noblerealms.json?style=_shields_test')
+  .expectJSON({ name: 'license', value: 'unknown', colorB: colorsB.lightgrey });
+
+t.create('License with SPDX id not appearing in configuration')
+  .get('/license/user1/project-with-EFL-license.json?style=_shields_test')
+  .intercept(nock => nock('https://api.github.com')
+    .get(/\/repos\/user1\/project-with-EFL-license/)
+    // GitHub API currently returns "other" as a key for repo with EFL license
+    .reply(200, `
```

Comment:
I'm 80% sure `nock` will let you put an object literal here instead of a string literal, and it'll still work.

Reply:
Thanks! Changed :-)